app/main.py

```diff
@@ -1,2 +1,96 @@
+from typing import Any, Hashable
+
+
+class Node:
+    def __init__(self, key: Hashable, hash_key: int, value: Any) -> None:
+        self.key = key
+        self.hash_key = hash_key
+        self.value = value
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            initial_capacity: int = 8,
+            load_factor: float = 0.75
+    ) -> None:
+        self._capacity = initial_capacity
+        self._load_factor = load_factor
+        self._size = 0
+        self._buckets = [[] for _ in range(self._capacity)]
+
+    def _get_index(self, key: Hashable) -> int:
+        return hash(key) % self._capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self._size / self._capacity >= self._load_factor:
+            self._resize()
+        index = self._get_index(key)
+        for node in self._buckets[index]:
+            if node.key == key:
+                node.value = value
+                return
+        self._buckets[index].append(Node(key, hash(key), value))
```

Comment:
you should implement your table differently - your node should be assigned directly at that index, not added to the list